### PR TITLE
DDF-3941 - Move intrigue from using the the Authentication Endpoint under /services

### PR DESCRIPTION
**Pull request by blen-desta**
#### What does this PR do?
**This is PR 6/7**

- Extracted the implementation of the AuthenticationEndpoint into a service
- Created new endpoint under `/search/catalog/internal`
- Changed intrigue code under `catalog-search-ui` to use new endpoint

#### Who is reviewing it? 
@brjeter @bakejeyner 

#### Ask 2 committers to review/merge the PR and tag them here. If you don't know who to ask, you can request reviews in https://groups.google.com/forum/#!forum/ddf-developers .
(please choose ONLY two committers from below, delete the rest)
@coyotesqrl 
@stustison

#### How should this be tested? (List steps with links to updated documentation)
- Set the authentication method for `/search` to `GUEST`
- Go to Intrigue
- Login from Intrigue

#### What are the relevant tickets?
[DDF-3941](https://codice.atlassian.net/browse/DDF-3941)

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

File: features/security/src/main/feature/feature.xml [hunk 1]
```diff
@@ -133,7 +133,9 @@
         <feature>security-handler-api</feature>
         <feature>security-policy-context</feature>
         <feature>security-core</feature>
-        <bundle>mvn:ddf.platform.security/security-rest-authentication/${project.version}</bundle>
+        <bundle>mvn:ddf.platform.security/security-rest-authn-service/${project.version}</bundle>
```
Comment (coyotesqrl) [inline features/security/src/main/feature/feature.xml]: ❓ Is it worth collecting these three bundles into a feature?

File: catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/AuthenticationApplication.java [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+    post(
+        "/login",
+        (req, res) -> {
+          Map<String, String> formParams = new HashMap<>();
+
+          // Extract form parameters
+          String body = req.body();
+          Pattern regExp =
+              Pattern.compile(
+                  "^Content-Disposition: form-data; name=\"(.+)\"$[\r\n]+(.+)$", Pattern.MULTILINE);
```
Comment (coyotesqrl) [inline catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/AuthenticationApplication.java]: ❓ I'm unclear about the use of this regex. Is this the standard/approved way of extracting multipart data with Spark? It seems rather brittle.

Comment (rzwiefel) [reply catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/security/AuthenticationApplication.java]: Looks we can do stuff like this. Do any of these not work @blen-desta ?

https://stackoverflow.com/questions/29373468/spark-java-how-to-handle-multipart-form-data-input

https://www.firstfewlines.com/post/spark-java-file-upload-example/

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
Successful build not reported correctly
build now

**Comment by blen-desta** [comment]
build now

**Comment by blen-desta** [comment]
Tests pass locally
Rebased against OWASP fix
build now

**Pull request merged**